### PR TITLE
perf: avoid stringify when set header

### PR DESCRIPTION
**Pull request by dead-horse**

### CODE REVIEW

File: lib/response.js [hunk 1]
```diff
@@ -439,8 +439,8 @@ module.exports = {
     if (this.headerSent) return;
 
     if (2 == arguments.length) {
-      if (Array.isArray(val)) val = val.map(String);
-      else val = String(val);
+      // maybe we can drop support of undefined in the next major version
+      if (val === undefined) val = 'undefined';
```
Comment (dead-horse) [inline lib/response.js]: other type of values will be stringify in `res.setHeader`

**Comment by dead-horse** [comment]
about ~3% performance improve.

-before 
image

- after

image

File: lib/response.js (hunk 1 above)
Comment (fengmk2) [reply lib/response.js]: res.getHeader will broken change after this change.

Comment (fengmk2) [reply lib/response.js]: > Sets a single header value for headers object. If this header already exists in the to-be-sent headers, its value will be replaced. Use an array of strings here to send multiple headers with the same name. Non-string values will be stored without modification. Therefore, request.getHeader() may return non-string values. However, the non-string values will be converted to strings for network transmission.

Comment (dead-horse) [reply lib/response.js]: fixed

**Pull request merged**